concent_api/core/migrations/0003_add_and_update_deadline_and_size.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-06-19 09:37
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db import models
+from common.helpers import deserialize_message
+from common.helpers import parse_timestamp_to_utc_datetime
+
+
+def populate_subtask_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(result_package_size__isnull=True):
+        stored_message = StoredMessage.objects.filter(
+            pk=subtask.report_computed_task_id
+        ).first()
+        if stored_message is not None:
+            subtask.result_package_size = deserialize_message(
+                stored_message.data
+            ).size
+            subtask.full_clean()
+            subtask.save()
+
+
+def populate_deadline_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(computation_deadline__isnull=True):
+        stored_message = StoredMessage.objects.filter(
+            pk=subtask.task_to_compute_id
+        ).first()
+        if stored_message is not None:
+            subtask.computation_deadline = parse_timestamp_to_utc_datetime(
+                deserialize_message(
+                    stored_message.data
+                ).compute_task_def['deadline']
```

Comment:
`StoredMessage` can store various types of messages. You'll get not only `TaskToCompute` but all kinds of other messages associated with the task. So the `deadline` field may not be present. It's only in `TaskToCompute`.

Actually, I think that it would be better if `computation_deadline` were `None` on `StoredMessage` instances that do not represent `TaskToCompute`.

Reply:
But I make those fields in Subtask model not in StoredMessage. TaskToCompute is always present in Subtask

Reply:
OK. But the name was `stored_message`. That's why you should avoid using misleading names :)